napari/layers/shapes/_shapes_models/shape.py

```diff
@@ -120,76 +120,76 @@ def __init__(
 
     @property
     @abstractmethod
-    def data(self):
+    def data(self) -> npt.NDArray:
         # user writes own docstring
         raise NotImplementedError
 
     @data.setter
     @abstractmethod
-    def data(self, data):
+    def data(self, data: npt.NDArray) -> None:
         raise NotImplementedError
 
     @abstractmethod
     def _update_displayed_data(self) -> None:
         raise NotImplementedError
 
     @property
-    def ndisplay(self):
+    def ndisplay(self) -> Literal[2]:
         """int: Number of displayed dimensions."""
         return self._ndisplay
 
     @ndisplay.setter
-    def ndisplay(self, ndisplay):
+    def ndisplay(self, ndisplay: Literal[2]) -> None:
         if self.ndisplay == ndisplay:
             return
         self._ndisplay = ndisplay
         self._update_displayed_data()
 
     @property
-    def dims_order(self):
+    def dims_order(self) -> list[int]:
         """(D,) list: Order that the dimensions are rendered in."""
         return self._dims_order
 
     @dims_order.setter
-    def dims_order(self, dims_order):
+    def dims_order(self, dims_order: list[int]) -> None:
         if self.dims_order == dims_order:
             return
         self._dims_order = dims_order
         self._update_displayed_data()
 
     @property
-    def dims_displayed(self):
-        """tuple: Dimensions that are displayed."""
+    def dims_displayed(self) -> list[int]:
+        """list: Dimensions that are displayed."""
         return self.dims_order[-self.ndisplay :]
 
     @property
-    def dims_not_displayed(self):
-        """tuple: Dimensions that are not displayed."""
```

Comment:
All these things *should* be tuples, ideally, I think...

Reply:
This seems to be the only property in `napari` called `dims_not_displayed`, so I couldn't find any other example. It doesn't seem to be used by anything either, so I could coerce it into a tuple if that's better?